teps/0060-remote-resource-resolution.md

```diff
@@ -242,207 +251,533 @@ users can leverage those shared pipelines.
+spec:
+  params:
+    image_url: gcr.io/tekton-releases/catalog/upstream/golang-fuzz:0.1
+    name: golang-fuzz
+```
+
+A newly-created `ResourceRequest` for a task from the catalog git repo:
+
+```yaml
+apiVersion: resolution.tekton.dev/v1alpha1
+kind: ResourceRequest
+metadata:
+  name: get-task-from-git-12345
+  namespace: quux
+  labels:
+    resolution.tekton.dev/resolver: git
+  ownerReferences:
+  - apiVersion: tekton.dev/v1beta1
+    controller: true
+    kind: TaskRun
+    name: my-tr
+    uid: 6aa5857a-3d67-4a09-94a1-8e9cc136dcf8
+spec:
+  params:
+    repository_url: https://github.com/tektoncd/catalog.git
+    branch: main
+    path: /task/golang-fuzz/0.1/golang-fuzz.yaml
+```
+
+A successfully completed `ResourceRequest`:
+
+```yaml
+apiVersion: resolution.tekton.dev/v1alpha1
+kind: ResourceRequest
+metadata:
+  name: get-pipeline-from-git-12345
+  namespace: quux
+  labels:
+    resolution.tekton.dev/resolver: git
+  ownerReferences:
+  - apiVersion: tekton.dev/v1beta1
+    controller: true
+    kind: PipelineRun
+    name: my-pr
+    uid: 6aa5857a-3d67-4a09-94a1-8e9cc136dcf8
+spec:
+  params:
+    repository_url: https://github.com/sbwsg/experimental.git
+    commit: 2c9b093e94f30f588dc798cc56a2559d4da9d573
+    path: /remote-resolution/pipeline.yaml
+status:
+  annotations:
+    commit: 2c9b093e94f30f588dc798cc56a2559d4da9d573
+    content-type: application/x-yaml
+  data: a2luZDogUGlwZWxpbmUKYXBpVmVyc2lvbjogdGVrdG9uLmRldi92MWJldGExCm1ldGFkYXRhOgogIG5hbWU6IHAKc3BlYzoKICB0YXNrczoKICAtIG5hbWU6IGZldGNoLWZyb20tZ2l0CiAgICB0YXNrU3BlYzoKICAgICAgc3RlcHM6CiAgICAgIC0gaW1hZ2U6IGFscGluZS9naXQKICAgICAgICBzY3JpcHQ6IHwKICAgICAgICAgIGdpdCBjbG9uZSBodHRwczovL2dpdGh1Yi5jb20vdGVrdG9uY2QvcmVzdWx0cwo=
+  conditions:
+  - lastTransitionTime: "2021-10-14T15:38:14Z"
+    status: "True"
+    type: Succeeded
+```
+
+A failed `ResourceRequest`:
+
+```yaml
+apiVersion: resolution.tekton.dev/v1alpha1
+kind: ResourceRequest
+metadata:
+  name: get-pipeline-from-git-12345
+  namespace: quux
+  labels:
+    resolution.tekton.dev/resolver: git
+  ownerReferences:
+  - apiVersion: tekton.dev/v1beta1
+    controller: true
+    kind: PipelineRun
+    name: my-pr
+    uid: 6aa5857a-3d67-4a09-94a1-8e9cc136dcf8
+spec:
+  params:
+    repository_url: https://github.com/sbwsg/experimental.git
+    branch: remote-resolution
+    path: /remote-resolution/invalid-pipeline.yaml
+status:
+  conditions:
+  - lastTransitionTime: "2021-10-19T11:06:00Z"
+    message: 'error opening file "/remote-resolution/invalid-pipeline.yaml": file does not exist'
+    reason: ResolutionFailed
+    status: "False"
+    type: Succeeded
+```
+
+### Resolver specifics
+
+Each Resolver runs as its own controller in the cluster. This allows an
+operator to spin up or tear down support for individual resolvers by
+`apply`ing or `delete`ing them.
+
+A resolver observes `ResourceRequest` objects, filtering on the
+`resolution.tekton.dev/resolver` label to find only those it is
```

Comment:
I like this idea a lot. It reminds me of the use of `ClusterInterceptor` in Triggers.

I am wondering if we can sneak up on this requirement during alpha rather than introduce an additional CRD in this TEP though. For example: what if we added a specific label or annotation on resolver deployments in the `tekton-remote-resolution` namespace? This way an interested party (like the Pipelines controller) who has read access to `tekton-remote-resolution` can query for the available resolvers but we don't need to pay the cost of specifying an additional custom resource type until we know it's useful and necessary?

Reply:
SGTM